src/functions/docker-exec-util.go

```diff
@@ -0,0 +1,235 @@
+package functions
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	network "github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	natting "github.com/docker/go-connections/nat"
+)
+
+func ExecCommand(client *client.Client, containerId string, commands []string) error {
+
+	createdExcComand := strings.Join(commands, " && ")
+	c := types.ExecConfig{
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          []string{"sh", "-c", createdExcComand},
+		Tty:          false,
+		Detach:       false,
+	}
+	execID, _ := client.ContainerExecCreate(context.Background(), containerId, c)
+	fmt.Println(execID)
+
+	res, err := client.ContainerExecAttach(context.Background(), execID.ID, types.ExecStartCheck{
+		Detach: false,
+		Tty:    false,
+	})
+	if err != nil {
+		return err
+	}
+	defer res.Close()
+
+	err = client.ContainerExecStart(context.Background(), execID.ID, types.ExecStartCheck{})
+	if err != nil {
+		return err
+	}
+
+	run := true
+	for run {
+		resp, err := client.ContainerExecInspect(context.Background(), execID.ID)
+		if err != nil {
+			panic(err)
+		}
+
+		if !resp.Running {
+			run = false
+		}
+		time.Sleep(250 * time.Millisecond)
+	}
+
+	return nil
+}
+
+func RunContainer(client *client.Client, imagename string, containername string, port string, inputEnv []string) (string, error) {
+	// Define a PORT opening
+	newport, err := natting.NewPort("tcp", port)
+	if err != nil {
+		fmt.Println("Unable to create docker port")
+		return "", err
+	}
+
+	// Configured hostConfig:
+	// https://godoc.org/github.com/docker/docker/api/types/container#HostConfig
+	hostConfig := &container.HostConfig{
+		PortBindings: natting.PortMap{
+			newport: []natting.PortBinding{
+				{
+					HostIP:   "0.0.0.0",
+					HostPort: port,
+				},
+			},
+		},
+		RestartPolicy: container.RestartPolicy{
+			Name: "always",
+		},
+		LogConfig: container.LogConfig{
+			Type:   "json-file",
+			Config: map[string]string{},
+		},
+	}
+
+	// Define Network config (why isn't PORT in here...?:
+	// https://godoc.org/github.com/docker/docker/api/types/network#NetworkingConfig
+	networkConfig := &network.NetworkingConfig{
+		EndpointsConfig: map[string]*network.EndpointSettings{},
+	}
+	gatewayConfig := &network.EndpointSettings{
+		Gateway: "gatewayname",
+	}
+	networkConfig.EndpointsConfig["bridge"] = gatewayConfig
+
+	// Define ports to be exposed (has to be same as hostconfig.portbindings.newport)
+	exposedPorts := map[natting.Port]struct{}{
+		newport: {},
+	}
+
+	// Configuration
+	// https://godoc.org/github.com/docker/docker/api/types/container#Config
+	config := &container.Config{
+		Image:        imagename,
+		Env:          []string{"DB_HOST=192.168.225.51", "DB_PORT=9876"},
```

Comment:
I would prefer we not hardcode addresses

Reply:
eliminated hardcoded values